src/app/select-session/page.tsx

```diff
@@ -0,0 +1,228 @@
+        cache: "no-store",
+      });
+      return await response.json();
+    }})
+
+  const queriedSessions : SessionCardProps[] | undefined = data?.map((session: GameSession) => {
+    const startDate = session.dateTime
+    const endDate = session.dateTime
+    endDate.setHours(endDate.getHours() + 1); //TODO: confirm
+    return {
+      startDate,
+      endDate,
+      location: session.location,
+      status: SessionCardStatus.DEFAULT,
+  }})
+
+  const sessions: SessionCardProps[] = [
+    {
+      startDate: new Date("2023-05-10T10:30:00"),
+      endDate: new Date("2023-05-10T11:30:00"),
+      location: "Auckland Badminton Association",
+      status: SessionCardStatus.DEFAULT,
+    },
+    {
+      startDate: new Date("2023-05-10T10:30:00"),
+      endDate: new Date("2023-05-10T11:30:00"),
+      location: "Auckland Badminton Association",
+      status: SessionCardStatus.DISABLED,
+    },
+    {
+      startDate: new Date("2023-05-10T10:30:00"),
+      endDate: new Date("2023-05-10T11:30:00"),
+      location: "Auckland Badminton Association",
+      status: SessionCardStatus.DEFAULT,
+    },
+    {
+      startDate: new Date("2023-05-10T10:30:00"),
+      endDate: new Date("2023-05-10T11:30:00"),
+      location: "Auckland Badminton Association",
+      status: SessionCardStatus.DEFAULT,
+    },
+    {
+      startDate: new Date("2023-05-10T10:30:00"),
+      endDate: new Date("2023-05-10T11:30:00"),
+      location: "Auckland Badminton Association",
+      status: SessionCardStatus.DEFAULT,
+    },
+  ];
+
+  const maxSessions: number = isMember ? 2 : 1;
+  const [isOverflown, setIsOverflown] = useState(false);
+
+  const [session, setSession] = useState<SessionCardProps[]>(sessions);
+  const [sessionsSelected, setSessionsSelected] = useState(0);
+  const [shake, setShake] = useState(false);
+  const [scrollIndicator, setScrollIndicator] = useState(true);
+  const ref = useRef<HTMLDivElement>(null);
+
+  /**Changes the status of the card on click conditionally
+   * If number of cards active exceeds the allowed amount, plays error animation
+   */
+  function sessionClick(e: ChangeEvent<HTMLInputElement>, index: number) {
+    if (e.target.checked && sessionsSelected === maxSessions) {
+      e.currentTarget.checked = false;
+      setShake(true);
+      setTimeout(() => {
+        setShake(false);
+      }, 480);
+    } else {
+      let tempArray = [...session];
+      tempArray[index].status = e.target.checked
+        ? SessionCardStatus.SELECTED
+        : SessionCardStatus.DEFAULT;
+      setSession(tempArray);
+    }
+  }
+
+  /**On status change, changes the counter of number of sessions selected*/
+  useEffect(() => {
+    const numberActive = session.filter((card: SessionCardProps) => {
+      if (card.status === SessionCardStatus.SELECTED) {
+        return true;
+      } else {
+        return false;
+      }
+    }).length;
+    setSessionsSelected(numberActive);
+  }, [session]);
```

Comment:
I don't think this should be in a useEffect. useEffect is more for synchronisation not events.

Explained in this long talk if your keen to watch
https://www.youtube.com/watch?v=CFxA5GmDIbU